feat: 🚀  Integrate Amazon Bedrock Support2

Sunwood-ai-labs opened the pull request:
## Overview
This PR adds support for Amazon Bedrock models to our LLM integration, enhancing our AI capabilities with Claude 3 models (Opus, Sonnet, Haiku).

## Key Changes
- Implement AWS credentials retrieval for Bedrock
- Add Bedrock model initialization and handling
- Include Claude 3 models options for Bedrock
- Adjust token limits specifically for Bedrock models
- Update chat action to support dynamic model selection
- Add necessary dependencies for Amazon Bedrock integration

## Detailed Changes
1. `app/lib/.server/llm/api-key.ts`: 
   - Add `getAWSCredentials` function to fetch AWS access keys and region

2. `app/lib/.server/llm/constants.ts`: 
   - Define `MAX_TOKENS_BEDROCK` constant (4096) for Bedrock models

3. `app/lib/.server/llm/model.ts`: 
   - Implement `getBedrockModel` function for Bedrock model initialization
   - Update `getModel` function to handle Bedrock provider

4. `app/lib/.server/llm/stream-text.ts`: 
   - Use Bedrock-specific token limit (`MAX_TOKENS_BEDROCK`)

5. `app/routes/api.chat.ts`: 
   - Update chat action to support model selection in the request body
   - Set proper Content-Type header for the response

6. `app/utils/constants.ts`: 
   - Add Bedrock model options (Claude 3 Opus, Sonnet, Haiku)

7. `package.json`: 
   - Add `@ai-sdk/amazon-bedrock` dependency (version 0.0.30)

8. `pnpm-lock.yaml`: 
   - Update with new dependencies including AWS SDK packages

coleam00 committed:
Added the ability to use practically any LLM you can dream of within Bolt.new

coleam00 committed:
Added the OpenRouter provider and a few models from OpenRouter (easily extendable to include more!)

Sunwood-ai-labs committed:
🚀 [feat] Integrate Amazon Bedrock support

Add support for Amazon Bedrock models, including:

- Implement AWS credentials retrieval for Bedrock
- Add Bedrock model initialization and handling
- Include Claude 3 models (Opus, Sonnet, Haiku) for Bedrock
- Adjust token limits for Bedrock models
- Update chat action to support model selection
- Add @ai-sdk/amazon-bedrock dependency

Key changes:
- app/lib/.server/llm/api-key.ts: Add getAWSCredentials function
- app/lib/.server/llm/constants.ts: Define MAX_TOKENS_BEDROCK
- app/lib/.server/llm/model.ts: Implement getBedrockModel function
- app/lib/.server/llm/stream-text.ts: Use Bedrock-specific token limit
- app/routes/api.chat.ts: Update to support model selection
- app/utils/constants.ts: Add Bedrock model options
- package.json: Add @ai-sdk/amazon-bedrock dependency
- pnpm-lock.yaml: Update with new dependencies

Sunwood-ai-labs committed:
🔑 [refactor] Improve API key and AWS credentials management

- Translate comments to English for consistency
- Add explanatory comment for AWS credentials function
- Refactor default region assignment with inline comment

Sunwood-ai-labs commented:
I changed the branch name, but the content is the same as  - #689!

Sunwood-ai-labs committed:
📚 [docs] Remove fork-specific information from README

- Deleted references to fork by Cole Medin
- Removed information about choosing LLM models
- Maintained focus on original Bolt.new project description

rrfaria commented on README.md:
````diff
@@ -21,34 +21,80 @@ Whether you’re an experienced developer, a PM or designer, Bolt.new allows you
 
-- **Batch simple instructions**: Save time by combining simple instructions into one message. For example, you can ask Bolt to change the color scheme, add mobile responsiveness, and restart the dev server, all in one go saving you time and reducing API credit consumption significantly.
+```bash
+git clone https://github.com/coleam00/bolt.new-any-llm.git
````
repository name is pointing to coleam00's github

Sunwood-ai-labs replied:
I responded with  [e141171](https://github.com/stackblitz/bolt.new/pull/733/commits/e14117113250213e2ed3db939900fd4d211384af)

rrfaria commented on app/lib/.server/llm/model.ts:
```diff
@@ -1,9 +1,77 @@
-  return anthropic('claude-3-5-sonnet-20240620');
+  return anthropic(model);
+}
+
```
using openRoute you can use gemini but if you want to use gemini directly you need to add gemini provider
```js
export function getGeminiModel(apiKey: string, modelName: string = 'gemini-1.5-pro-latest') {
  const model = createGoogleGenerativeAI({
    apiKey,
  });

  return model(modelName);
}
```

Sunwood-ai-labs replied:
I responded with [e5d16df](https://github.com/stackblitz/bolt.new/pull/733/commits/e5d16dffd885ed9ba413c3423dff1cf4dbb53636)

rrfaria commented on app/lib/.server/llm/model.ts:
```diff
@@ -1,9 +1,77 @@
+    case 'Anthropic':
+      return getAnthropicModel(apiKey, model);
+    case 'OpenAI':
+      return getOpenAIModel(apiKey, model);
```
also add gemini Provider here.

Sunwood-ai-labs replied:
I responded with [e5d16df](https://github.com/stackblitz/bolt.new/pull/733/commits/e5d16dffd885ed9ba413c3423dff1cf4dbb53636)

rrfaria commented on app/lib/.server/llm/api-key.ts:
```diff
@@ -1,9 +1,32 @@
+  switch (provider) {
+    case 'Anthropic':
+      return env.ANTHROPIC_API_KEY || cloudflareEnv.ANTHROPIC_API_KEY;
+    case 'OpenAI':
```
add api key from gemini provider

Sunwood-ai-labs replied:
I responded with [7e0287f](https://github.com/stackblitz/bolt.new/pull/733/commits/7e0287f7f6198497ae93a76759a5e83f653fafe4)

Sunwood-ai-labs committed:
🔑 [feat] Add Gemini API key support

- Include Gemini API key in the getAPIKey function
- Allow retrieval of Gemini API key from environment variables

Sunwood-ai-labs committed:
🤖 [feat] Implement Gemini model integration

- Import Google Generative AI SDK
- Add getGeminiModel function to create Gemini model instances
- Update getModel function to support Gemini provider

Sunwood-ai-labs committed:
📋 [feat] Add Gemini models to MODEL_LIST

- Include Gemini 1.5 Pro and Flash models in the available model options
- Add latest and stable versions for both Gemini 1.5 Pro and Flash

Sunwood-ai-labs committed:
📦 [chore] Add @ai-sdk/google dependency

- Include @ai-sdk/google package version 0.0.52 for Gemini integration

Sunwood-ai-labs committed:
🔒 [chore] Update pnpm-lock.yaml with new dependencies

- Add @ai-sdk/google package and its dependencies to the lock file
- Ensure consistent package versions across the project

Sunwood-ai-labs committed:
📝 [docs] Update repository URL in clone instructions

- Replace old URL (https://github.com/coleam00/bolt.new-any-llm.git) with new URL (https://github.com/stackblitz/bolt.new)
- Improve documentation accuracy for users setting up the project

rrfaria commented on worker-configuration.d.ts:
```diff
@@ -1,3 +1,6 @@
 interface Env {
   ANTHROPIC_API_KEY: string;
+  OPENAI_API_KEY: string;
+  GROQ_API_KEY: string;
```
One last thing to approve:
add GEMINI_API_KEY here and also on .env.example

Sunwood-ai-labs replied:
thank you！
I responded with  [9e5e73a](https://github.com/stackblitz/bolt.new/pull/733/commits/9e5e73a0096b50aae64703d8823ec97aec00dea9)

Sunwood-ai-labs committed:
🔐 [feat] Add AWS credentials configuration

- Added AWS credential environment variables:
  - AWS_ACCESS_KEY_ID
  - AWS_SECRET_ACCESS_KEY
  - AWS_REGION
- Enables AWS service integration capabilities

AriPerkkio reviewed (commented):
Thanks for the PR @Sunwood-ai-labs.

The `bolt.new` is currently not accepting contributions. However there is https://bolt.diy/ that welcomes developers and builders to contribute. Check out https://www.youtube.com/watch?v=aZn8PhqUZVU and https://thinktank.ottomator.ai/.
